dashboard/src/shared/auth/AuthnContext.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Redirect, Route, Switch } from "react-router-dom";
+
+import api from "shared/api";
+import { Context } from "shared/Context";
+
+import Loading from "../../components/Loading";
+import Login from "../../main/auth/Login";
+import Register from "../../main/auth/Register";
+import ResetPasswordFinalize from "../../main/auth/ResetPasswordFinalize";
+import ResetPasswordInit from "../../main/auth/ResetPasswordInit";
+import SetInfo from "../../main/auth/SetInfo";
+import VerifyEmail from "../../main/auth/VerifyEmail";
+
+type AuthnState = {
+  userId: number;
+  handleLogOut: () => void;
+};
+
+const NilAuthnState = {
+    userId: -1,
+    handleLogOut: () => {},
+};
+
+export const AuthnContext = React.createContext<AuthnState>(NilAuthnState);
+
```

Comment:
I'd maybe just use the pattern that we have in the other context providers, which will just error if the values haven't been set yet.

```suggestion

export const AuthnContext = React.createContext<AuthnState | null>(null);

export const useAuthState = (): AuthnSate => {
  const context = useContext(AuthnContext)
  if (context == null) {
    throw new Error(
      "useAuthState must be used within a AuthnContext"
    );
  }
  return context;
}
```